### PR TITLE
ddl: add virtual columns compatibility for temporary table

### PR DESCRIPTION
**Pull request by Howie59**
### What problem does this PR solve?

Issue Number: see https://github.com/pingcap/tidb/pull/24168

### What is changed and how it works?

Proposal: [temporary table](https://github.com/pingcap/tidb/issues/24169) 

What's Changed:  add virtual columns compatibility for temporary table

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -1814,6 +1814,13 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if tbInfo.TempTableType != model.TempTableNone {
+		for _, column := range tbInfo.Columns {
+			if column.IsGenerated() && !column.GeneratedStored {
+				return ErrOptOnTemporaryTable.GenWithStackByArgs("virtual columns")
```
Comment (djshow832) [inline ddl/ddl_api.go:1820]: Is it really unable to support virtual columns? Would you have a try?

Comment (djshow832) [reply ddl/ddl_api.go:1820]: If it's able to support it, we don't need to forbid it.

Comment (Howie59) [reply ddl/ddl_api.go:1820]: i test it on my compute, if the column is virtual, it match the condition in line 1819. why don't need to forbid it?

**Review by djshow832 — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**